File: dynamic_listings/python/default/gameloop/semifixed_timesteps.py
import time

dt: float = 1.0/60.0
game_is_running: bool = True

# We bootstrap frametime for 1/60th of a second for the first frame
frame_time: float = 1.0/60.0

while game_is_running:
    # We get the system time in milliseconds
    begin: float = time.time() * 1000

    while frame_time > 0.0:
        delta_time: float = min(dt, frame_time)
        process_user_input()
        update_world(dt)
        frame_time -= delta_time
    draw()
    end: float = time.time() * 1000
    # We memorize how long this frame lasted
    frame_time = end - begin

File: dynamic_listings/C++/default/gameloop/semifixed_timesteps.cpp
float dt = 1.0/60.0;
bool game_is_running = true;

// We bootstrap frametime for 1/60th of a second for the first frame
float frametime = 1.0/60.0;

while(game_is_running){
    // We get the system time in milliseconds
    // since implementation varies here i'll use a generic function name
    float begin = get_system_time_millis();

    while(frametime > 0.0){
        float deltaTime = min(dt, frametime);
        process_user_input();
        update_world(dt);
        frametime = frametime - deltaTime;
        draw();
    }
    float end = get_system_time_millis();
    // We memorize how long this frame lasted
    frametime = end - begin;
}
